### PR TITLE
Bail on payment errors, update cli putput for failed chunks

### PR DESCRIPTION
**Pull request by joshuef**
### Summary generated by Reviewpad on 08 Dec 23 09:09 UTC
This pull request includes the following changes:

1. The file `CHANGELOG.md` has been modified in the following ways:
- Updated dependencies in the `Other` section
- Added a new version entry `0.2.328` with an update to dependencies in the `Other` section

2. The file `sn_node/CHANGELOG.md` has been modified with the following changes:
- Added a new version [0.99.12](https://github.com/maidsafe/safe_network/compare/sn_node-v0.99.11...sn_node-v0.99.12) with updates to dependencies.

3. The file `lib.rs` in the `sn_peers_acquisition` module has been modified with the addition of an attribute `#[allow(clippy::unused_async)]` and a function `get_network_contacts` that is conditionally compiled based on the absence of the feature "network-contacts". The function returns an empty vector of `Multiaddr` and is marked as async.

4. A command in the GitHub workflow file `nightly.yml` has been updated. Specifically, the command `cargo run --bin safe --release -- --log-output-dest=data-dir register create baobao` has been modified to `cargo run --bin safe --release -- --log-output-dest=data-dir register create -n baobao` by adding the `-n` flag before the argument `baobao`. This change is made for creating a register in the client.

5. The file `nodes_rewards.rs` has undergone several changes including the addition of a new variable, modification of the rewards calculation and verification logic, adjustment of sleep durations, and modification of loop conditions for better functionality.

6. The file `Cargo.toml` has been modified with a version update from `0.86.47` to `0.86.48`.

7. The `Cargo.toml` file in the `sn_faucet` directory has been updated with a version bump from `0.1.69` to `0.1.70`.

8. The file `Cargo.lock` has been modified with updates to the versions of various packages including `sn_cli`, `sn_faucet`, `sn_node`, `sn_node_rpc_client`, and `sn_testnet`.

9. The diff of the `Cargo.toml` file in the `sn_testnet` directory shows a version update from `0.2.327` to `0.2.328`.

10. The file `CHANGELOG.md` has been modified with the addition of a new entry for version `0.1.66` which includes updates to dependencies.

11. The diff of the file `sn_cli/src/subcommands/files/mod.rs` includes various changes related to error handling, logging, and display of file information and costs.

12. The file diff includes changes to the `CHANGELOG.md` file in the `sn_faucet` directory, including the addition of a new section for version 0.1.70 and an update to dependencies.

13. The GitHub workflow file `.github/workflows/merge.yml` has been updated with a version change in the action `wagoid/commitlint-github-action`.

14. The `Cargo.toml` file in the `sn_node_rpc_client` directory has been modified with version updates and changes in dependencies.

15. The file `Cargo.toml` has a version change from `0.99.11` to `0.99.12`.

16. The file `sn_cli/CHANGELOG.md` includes the addition of a new release version `0.86.48` with updated dependencies.

Please let me know if you need more information or if there's anything else I can help you with!

### CODE REVIEW

**Comment by joshuef** [comment]
atop #1062

File: sn_cli/src/subcommands/files/mod.rs [hunk 1]
```diff
@@ -210,10 +211,30 @@ async fn upload_files(
...
     let now = Instant::now();
-    let mut recorded_pay_errors = vec![];
-    let mut recorded_upload_errors = vec![];
+
+    // Task set to add and remove chunks from the chunk manager
+    let mut uploading_chunks = FuturesUnordered::new();
 
     for chunks_batch in chunks_batches {
+        // while we dont have a full batch_size of ongoing uploading_chunks
+        // we can pay fo rthe next batch and carry on
```
Comment (grumbach) [inline sn_cli/src/subcommands/files/mod.rs]: `for the` typo

File: sn_networking/src/cmd.rs [hunk 2]
```diff
@@ -546,9 +558,6 @@ impl SwarmDriver {
 
                 let _ = sender.send(closest_peers);
             }
-            // SwarmCmd::GetOurCloseGroup { sender } => {
-            //     let _ = sender.send(self.close_group.clone());
-            // }
```
Comment (grumbach) [inline sn_networking/src/cmd.rs:551]: why keep it as a comment?

Comment (joshuef) [reply sn_networking/src/cmd.rs:551]: This removes the comment that was left some other time.

Comment (grumbach) [reply sn_networking/src/cmd.rs:551]: Oh right!! 💯

**Review by grumbach — APPROVED** [review]
LGTM

**Pull request merged**